apps/aehttp/test/aehttp_client.erl

```diff
@@ -2,67 +2,57 @@
 
-%% Required config values: int_addr, ext_addr
-%% Optional config values: encode_base58c
+%% Required config values: int_http, ext_http
+%% Optional config values: ct_log (true by default)
 request(OpId, Params, Cfg) ->
     Op = endpoints:operation(OpId),
     {Method, Interface} = operation_spec(Op),
-    BaseUrl = proplists:get_value(Interface, Cfg),
-    Params1 = maybe_encode_params(Params, Cfg),
-    Path = operation_path(Method, OpId, Params1),
-    request(Method, BaseUrl, Path, Params1, [], [], []).
+    BaseUrl = make_base_url(proplists:get_value(Interface, Cfg)),
+    Path = operation_path(Method, OpId, convert_params(Params)),
+    request(Method, BaseUrl, Path, Params, [], [], [], Cfg).
 
-request(get, BaseUrl, Path, QueryParams, Headers, HttpOpts, Opts) ->
+request(get, BaseUrl, Path, QueryParams, Headers, HttpOpts, Opts, Cfg) ->
     Url = make_url(BaseUrl, Path, QueryParams),
-    ct:log("GET ~p", [Url]),
+    log("GET ~p", [Url], Cfg),
     Resp = httpc:request(get, {Url, Headers}, HttpOpts, Opts),
-    process_response(Resp);
-request(post, BaseUrl, Path, BodyParams, Headers, HttpOpts, Opts) ->
+    process_response(Resp, Cfg);
+request(post, BaseUrl, Path, BodyParams, Headers, HttpOpts, Opts, Cfg) ->
     Url = make_url(BaseUrl, Path, #{}),
     {ContentType, Body} = make_body(BodyParams, Path),
-    ct:log("POST ~p~nContentType ~p~n~p", [Url, ContentType, BodyParams]),
+    log("POST ~p~nContentType ~p~n~p", [Url, ContentType, BodyParams], Cfg),
     Resp = httpc:request(post, {Url, Headers, ContentType, Body}, HttpOpts, Opts),
-    process_response(Resp).
+    process_response(Resp, Cfg).
 
 %%=============================================================================
 %% Internal functions
 %%=============================================================================
 
-process_response({ok, {{_, Code, _State}, _Head, Body}}) ->
-    try
-        case iolist_to_binary(Body) of
-            <<>> ->
-                ct:log("Return code: ~p", [Code]),
-                {ok, Code, #{}};
-            B ->
-                B1 = jsx:decode(B, [return_maps]),
-                ct:log("Return code: ~p, body: ~p", [Code, B1]),
-                {ok, Code, B1}
-        end
-    catch
-        error:E -> {error, {parse_error, [E, erlang:get_stacktrace()]}}
+process_response({ok, {{_, Code, _State}, _Head, Body}}, Cfg) ->
+    case iolist_to_binary(Body) of
+        <<>> ->
+            log("Return code: ~p", [Code], Cfg),
+            {ok, Code, #{}};
+        Body1 ->
+            Body2 = jsx:decode(Body1, [{labels, attempt_atom}, return_maps]),
+            log("Return code: ~p, body: ~p", [Code, Body2], Cfg),
+            {ok, Code, Body2}
     end;
-process_response({error, _} = Error) ->
+process_response({error, _} = Error, _Cfg) ->
     Error.
 
-maybe_encode_params(Params, Cfg) ->
-    case proplists:get_value(encode_base58c, Cfg) of
-        true -> encode_params(Params);
-        _ -> Params
+log(Fmt, Params, Cfg) ->
```

Comment:
Why not get a logging function?

Reply:
Do you mean passing log function in `Cfg`? That could be an option, too. But currently this client is going to be used in tests where there is only CT logging or no logging (QuichCheck tests).

Reply:
Just saying because this how system test actually works.